compute/cloud-client/src/test/java/compute/reservation/CreateReservationFromVmIT.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.reservation;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.compute.v1.Instance;
+import com.google.cloud.compute.v1.InstancesClient;
+import com.google.cloud.compute.v1.Reservation;
+import com.google.cloud.compute.v1.ReservationsClient;
+import compute.CreateInstance;
+import compute.DeleteInstance;
+import compute.Util;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class CreateReservationFromVmIT {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String ZONE = "us-east4-c";
+  private static ReservationsClient reservationsClient;
+  private static InstancesClient instancesClient;
+  private static String RESERVATION_NAME;
+  private static String INSTANCE_FOR_RESERVATION;
+  static String javaVersion = System.getProperty("java.version").substring(0, 2);
```

Comment:
Is Java version mandatory to test? Can this be replaced with an UUID?

Reply:
To avoid conflict for Java 11 and Java 17 in the cleanup method, I've added this javaVersion to the variable. We discussed it with Maciej and @minherz.